.gitignore

```diff
@@ -1,1071 +1,269 @@
 icu4c/source/extra/uconv/uconv
 icu4c/source/extra/uconv/uconv.1
 icu4c/source/extra/uconv/uconv.plg
-icu4c/source/extra/uconv/uconv.vcproj.*.*.user
 icu4c/source/extra/uconv/uconvmsg
-icu4c/source/extra/uconv/x64
-icu4c/source/extra/uconv/x86
-icu4c/source/extra/uconv/ARM
-icu4c/source/extra/uconv/ARM64
-icu4c/source/i18n/*.ao
-icu4c/source/i18n/*.d
-icu4c/source/i18n/*.o
-icu4c/source/i18n/*.plg
-icu4c/source/i18n/*.vcxproj.user
-icu4c/source/i18n/Debug
-icu4c/source/i18n/Generated[!!-~]Files
-icu4c/source/i18n/Makefile
-icu4c/source/i18n/Makefile.local
-icu4c/source/i18n/Release
-icu4c/source/i18n/arm
-icu4c/source/i18n/arm64
-icu4c/source/i18n/debug
 icu4c/source/i18n/i18n.res
-icu4c/source/i18n/i18n.vcproj.*.*.user
-icu4c/source/i18n/release
-icu4c/source/i18n/x64
-icu4c/source/i18n/x86
-icu4c/source/i18n/ARM
-icu4c/source/i18n/ARM64
 icu4c/source/icudefs.local
 icu4c/source/icudefs.mk
-icu4c/source/io/*.ao
-icu4c/source/io/*.d
-icu4c/source/io/*.o
-icu4c/source/io/*.vcxproj.user
-icu4c/source/io/Debug
-icu4c/source/io/Makefile
-icu4c/source/io/Release
-icu4c/source/io/debug
 icu4c/source/io/io.res
-icu4c/source/io/io.vcproj.*.*.user
-icu4c/source/io/release
-icu4c/source/io/x64
-icu4c/source/io/x86
-icu4c/source/io/ARM
-icu4c/source/io/ARM64
-icu4c/source/layoutex/*.ao
-icu4c/source/layoutex/*.d
-icu4c/source/layoutex/*.o
-icu4c/source/layoutex/*.pdb
-icu4c/source/layoutex/*.vcxproj.user
-icu4c/source/layoutex/.vs
-icu4c/source/layoutex/Debug
-icu4c/source/layoutex/Makefile
-icu4c/source/layoutex/Release
-icu4c/source/layoutex/debug
 icu4c/source/layoutex/layoutex.res
-icu4c/source/layoutex/layoutex.vcproj.*.*.user
-icu4c/source/layoutex/release
-icu4c/source/layoutex/x64
-icu4c/source/layoutex/x86
-icu4c/source/lib
-icu4c/source/perf-*.xml
-icu4c/source/samples/*.opensdf
-icu4c/source/samples/*.sdf
-icu4c/source/samples/Makefile
-icu4c/source/samples/all/*.opensdf
-icu4c/source/samples/all/*.sdf
-icu4c/source/samples/all/.vs
 icu4c/source/samples/all/all.ncb
 icu4c/source/samples/all/all.suo
-icu4c/source/samples/break/*.vcxproj.user
-icu4c/source/samples/break/Debug
 icu4c/source/samples/break/break
 icu4c/source/samples/break/break.exe
-icu4c/source/samples/break/break.out
-icu4c/source/samples/break/break.vcproj.*.*.user
-icu4c/source/samples/break/release
-icu4c/source/samples/break/x64
-icu4c/source/samples/break/x86
-icu4c/source/samples/cal/*.d
-icu4c/source/samples/cal/*.o
-icu4c/source/samples/cal/*.pdb
-icu4c/source/samples/cal/*.vcxproj.user
-icu4c/source/samples/cal/Debug
-icu4c/source/samples/cal/Makefile
-icu4c/source/samples/cal/Release
-icu4c/source/samples/cal/cal.vcproj.*.*.user
 icu4c/source/samples/cal/icucal
 icu4c/source/samples/cal/icucal.exe
-icu4c/source/samples/cal/release
-icu4c/source/samples/cal/x64
-icu4c/source/samples/cal/x86
-icu4c/source/samples/case/*.vcxproj.user
-icu4c/source/samples/case/Debug
 icu4c/source/samples/case/case
 icu4c/source/samples/case/case.exe
-icu4c/source/samples/case/case.out
```

Comment:
Did you mean to include `*.out` as one of the wildcard ignores?  (I'll stop flagging other .out files)